src/core/classes/storage.ts

```diff
@@ -0,0 +1,84 @@
+import { JsonObject } from "..";
+import { FileOrBuffer } from "../..";
+import { IStorage } from "../interfaces/IStorage";
+
+export class Storage {
+  private storage: IStorage;
+
+  constructor(storage: IStorage) {
+    this.storage = storage;
+  }
+
+  /**
+   * Fetch data from any IPFS hash without worrying about gateways, data types, etc.
+   * Simply pass in an IPFS url and we'll handle fetching for you and try every public gateway
+   * to get the fastest response.
+   *
+   * @example
+   * ```javascript
+   * // Your IPFS hash here
+   * const hash = "ipfs://..."
+   * const data = await sdk.storage.fetch(hash);
+   * ```
+   * @param hash - The IPFS hash of the file or data to fetch
+   * @returns The data stored at the specified IPFS hash
+   */
+  public async fetch(hash: string): Promise<Record<string, any>> {
+    return this.storage.get(hash);
+  }
+
+  /**
+   * Upload any data to an IPFS directory.
+   *
+   * @example
+   * ```javascript
+   * // Add all your data here
+   * const files = [
+   *   fs.readFileSync("file1.png"),
+   *   fs.readFileSync("file2.png"),
+   *   fs.readFileSync("file3.png"),
+   * ]
+   * const uri = await sdk.storage.upload(files);
+   * ```
+   *
+   * @param data - An array of file data or an array of JSON metadata to upload to IPFS
+   * @returns The IPFS hash of the directory that holds all the uploaded data
+   */
+  public async upload(
+    data: FileOrBuffer[] | JsonObject[] | FileOrBuffer | JsonObject,
+  ): Promise<string> {
```

Comment:
would still be nice to also return the individual uris here, or at least something like the total number of files uploaded?